Dockerfile.apps

```diff
@@ -0,0 +1,57 @@
+FROM gcc:10.2.0 AS builder
+
+RUN apt update \
+ && apt install --no-install-recommends -y \
+      cmake \
+      ninja-build \
+      iwyu \
+      python3-pip \
+      python3-setuptools \
+ && rm -rf /var/lib/apt/lists/*
+
+RUN pip3 install conan
+
+ARG CONAN_REMOTE_URL=http://localhost:9300/
+ARG CONAN_USERNAME=demo
+ARG CONAN_PASSWORD=demo
+
+RUN conan profile new default --detect \
+ && conan profile update settings.compiler.libcxx=libstdc++11 default \
+ && conan remote add --force ambrosys $CONAN_REMOTE_URL \
+ && conan user --remote ambrosys $CONAN_USERNAME --password $CONAN_PASSWORD
+
+COPY CMakeModules /src/CMakeModules
+COPY examples /src/examples
+COPY src /src/src
+COPY .git /src/.git
```

Comment:
Warum brauchen wir `.git`?

Reply:
Das wird noch im Code verwendet, CMake wertet `git describe` aus, um Versions- und Commit-Angaben in eine Header-Datei zu schreiben. Der C++-Code nutzt das aber derzeit nicht (mehr), kann man also überlegen zu entfernen, da es insbesondere in den CMake-Dateien eine hohe Komplexität erzeugt. Und auch dieses ominöse target `test-if-git-HEAD-has-changed` existiert nur deshalb. Aber das sollte in einem eigenen Ticket passieren.